modules/apps/29-fee/ibc_middleware.go

```diff
@@ -248,7 +248,7 @@ func (im IBCMiddleware) OnAcknowledgementPacket(
 		return im.app.OnAcknowledgementPacket(ctx, packet, acknowledgement, relayer)
 	}
 
-	var ack = &types.IncentivizedAcknowledgement{}
+	ack := &types.IncentivizedAcknowledgement{}
```

Comment:
Could we change this to be consistent with the rest of the codebase, we generally favour declare over initialise.

```suggestion
	var ack types.IncentivizedAcknowledgement
```

You'll need to update `UnmarshalJSON` to pass the reference `&ack` below

Reply:
@faddat I think the build is failing because you still need to do the code change that @damiannolan suggests above (pass `&ack` to `UnmarshallJSON`):

`
Error: modules/apps/29-fee/ibc_middleware.go:252:59: cannot use ack (variable of type "github.com/cosmos/ibc-go/v4/modules/apps/29-fee/types".IncentivizedAcknowledgement) as type "github.com/gogo/protobuf/proto".Message in argument to types.ModuleCdc.UnmarshalJSON:
[106](https://github.com/cosmos/ibc-go/runs/7388382589?check_suite_focus=true#step:5:107)
	"github.com/cosmos/ibc-go/v4/modules/apps/29-fee/types".IncentivizedAcknowledgement does not implement "github.com/gogo/protobuf/proto".Message (ProtoMessage method has pointer receiver)
`